### PR TITLE
Add PXF FDW code to the PXF repository

### PR DESCRIPTION
**Pull request by frankgh**
The PXF FDW code is moving to the PXF repository. The code is based on
the pxf_fdw_feature_6x version.

### CODE REVIEW

**Comment by oliverralbertini** [comment]
Do the pg_regress tests still run?

**Comment by frankgh** [comment]
@oliverralbertini yes they do
```
  fguerrero in ~/workspace/pxf/fdw
±  |pxf-fdw-c → origin {36} ?:18 ✗| → make installcheck
/usr/local/gpdb6/lib/postgresql/pgxs/src/makefiles/../../src/test/regress/pg_regress --inputdir=./ --psqldir='/usr/local/gpdb6/bin'    --dbname=contrib_regression pxf_fdw_wrapper pxf_fdw_server pxf_fdw_user_mapping pxf_fdw_foreign_table
(using postmaster on fguerrero, port 6000)
============== dropping database "contrib_regression" ==============
NOTICE:  database "contrib_regression" does not exist, skipping
DROP DATABASE
============== creating database "contrib_regression" ==============
CREATE DATABASE
ALTER DATABASE
============== checking optimizer status              ==============
Optimizer disabled. Using planner answer files
============== checking gp_resource_manager status    ==============
Resource group disabled. Using default answer files
============== running regression test queries        ==============
test pxf_fdw_wrapper          ... ok
test pxf_fdw_server           ... ok
test pxf_fdw_user_mapping     ... ok
test pxf_fdw_foreign_table    ... ok

=====================
 All 4 tests passed.
=====================
```

File: fdw/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+PXF is a query federation engine that accesses data residing in external systems
+such as Hadoop, Hive, HBase, relational databases, S3, Google Cloud Storage,
+among other external systems.
+
+### Development
+
+## Compile
+
+To compile the PXF foreign data wrapper, we need a Greenplum 6+ installation and libcurl.
+
```
Comment (oliverralbertini) [inline fdw/README.md]: Will this work with Greenplum 6? Don't we need some changes to the COPY code before that's the case?

Comment (frankgh) [reply fdw/README.md]: Great point, @oliverralbertini. We need that commit in 6. For now I have disabled that code, and I return an error. I can create a PR for that change on the GPDB project. Let me know how would you like me to proceed.

Comment (oliverralbertini) [reply fdw/README.md]: I think having it disabled for now is fine.

File: fdw/Makefile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+SHLIB_LINK += -lcurl
+
+EXTENSION = pxf_fdw
+DATA = pxf_fdw--1.0.sql
+
+REGRESS = pxf_fdw_wrapper pxf_fdw_server pxf_fdw_user_mapping pxf_fdw_foreign_table
+
+PG_CONFIG = pg_config
+PGXS := $(shell $(PG_CONFIG) --pgxs)
+include $(PGXS)
```
Comment (oliverralbertini) [inline fdw/Makefile]: comment from sync up: error out if not set?

Comment (frankgh) [reply fdw/Makefile]: done

**Pull request merged**